File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        for (int i = 1; i < n; i++) {
            if (nums[i] == 1) {
                nums[i] = nums[i - 1] + 1; 
            }
        }
        return *max_element(nums.begin(), nums.end()); // max consecutive ones return
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
     int count=0;
     int curr_count=0;
     for(int i=0;i<nums.length;i++){
        if(nums[i]==1){
            curr_count++;
            count=Math.max(curr_count,count);
        }
        else{
            curr_count=0;
        }
     } 

   return count; }
}